crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3815,6 +3863,7 @@ pub(crate) mod tests {
     #[test_case(Ty::Tuple(vec![]), Ty::BuiltinClassLiteral("object"))]
     #[test_case(Ty::SubclassOfBuiltinClass("object"), Ty::None)]
     #[test_case(Ty::SubclassOfBuiltinClass("str"), Ty::LiteralString)]
+    #[test_case(Ty::AlwaysFalsy, Ty::AlwaysTruthy)]
```

Comment:
Can we also test below in `is_not_disjoint_from` that e.g. `Instance("str")` is not disjoint from either `AlwaysTruthy` or `AlwaysFalsy`?